lib/services/platform-service.ts

```diff
@@ -484,32 +514,35 @@ export class PlatformService implements IPlatformService {
 		}).future<void>()();
 	}
 
-	private getFrameworkFiles(platformData: IPlatformData, version: string): IFuture<string[]> {
+	private getFrameworkFiles(platformData: IPlatformData, version: string): IFuture<any> {
 		return (() => {
-			var npmCacheDirectoryPath = this.getNpmCacheDirectory(platformData.frameworkPackageName, version).wait();
-			var allFiles = this.$fs.enumerateFilesInDirectorySync(npmCacheDirectoryPath);
+			var cachedPackagePath = this.$npm.getCachedPackagePath(platformData.frameworkPackageName, version);
+			this.ensurePackageIsCached(cachedPackagePath, platformData.frameworkPackageName, version).wait();
+
+			var allFiles = this.$fs.enumerateFilesInDirectorySync(cachedPackagePath);
 			var filteredFiles = _.filter(allFiles, file => _.contains(platformData.frameworkFilesExtensions, path.extname(file)));
-			var relativeToCacheFiles = _.map(filteredFiles, file => file.substr(npmCacheDirectoryPath.length));
 
-			return relativeToCacheFiles;
+			var allFrameworkDirectories = _.map(this.$fs.readDirectory(path.join(cachedPackagePath, constants.PROJECT_FRAMEWORK_FOLDER_NAME)).wait(), dir => path.join(cachedPackagePath, constants.PROJECT_FRAMEWORK_FOLDER_NAME, dir));
+			var filteredFrameworkDirectories = _.filter(allFrameworkDirectories, dir => this.$fs.getFsStats(dir).wait().isDirectory() && (_.contains(platformData.frameworkFilesExtensions, path.extname(dir)) || _.contains(platformData.frameworkDirectoriesNames, path.basename(dir))));
 
-		}).future<string[]>()();
+			return {
+				frameworkFiles: this.mapFrameworkFiles(cachedPackagePath, filteredFiles),
+				frameworkDirectories: this.mapFrameworkFiles(cachedPackagePath, filteredFrameworkDirectories)
+			}
+
+		}).future<any>()();
 	}
 
-	private getNpmCacheDirectory(packageName: string, version: string): IFuture<string> {
+	private ensurePackageIsCached(cachedPackagePath: string, packageName: string, version: string): IFuture<void> {
 		return (() => {
-			var npmCacheDirectoryPath = this.getNpmCacheDirectoryCore(packageName, version);
-
-			if(!this.$fs.exists(npmCacheDirectoryPath).wait()) {
+			if(!this.$fs.exists(cachedPackagePath).wait()) {
```

Comment:
I was thinking about it, but I can't figure out a better solution. I'm opened to any suggestions.